uberfire-security/uberfire-security-management/uberfire-security-management-backend/src/main/java/org/uberfire/ext/security/management/BackendUserSystemManager.java

```diff
@@ -0,0 +1,161 @@
+package org.uberfire.ext.security.management;
+
+import org.jboss.errai.security.shared.api.Group;
+import org.jboss.errai.security.shared.api.Role;
+import org.jboss.errai.security.shared.api.identity.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.ext.security.management.api.*;
+import org.uberfire.ext.security.management.api.validation.EntityValidator;
+import org.uberfire.ext.security.management.validation.GroupValidatorImpl;
+import org.uberfire.ext.security.management.validation.RoleValidatorImpl;
+import org.uberfire.ext.security.management.validation.UserValidatorImpl;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.inject.Inject;
+import java.util.Set;
+
+/**
+ * Note that role management services are not yet available. 
+ */
+@ApplicationScoped
+public class BackendUserSystemManager implements UserSystemManager {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BackendUserSystemManager.class);
+    
+    public static final String ENV_USERS_MANAGER_SERVICE = "org.uberfire.ext.security.management.api.usersManager";
+    public static final String ENV_GROUPS_MANAGER_SERVICE = "org.uberfire.ext.security.management.api.groupsManager";
+    public static final String ENV_DEFAULT_USERS_MANAGER_SERVICE = "WildflyUsersCLIManager";
+    public static final String ENV_DEFAULT_GROUPS_MANAGER_SERVICE = "WildflyGroupsCLIManager";
+    
+    @Inject
+    BeanManager beanManager;
+    
+    @Inject
+    Instance<UsersManager> usersManagerServices;
+    
+    @Inject
+    Instance<GroupsManager> groupsManagerServices;
+    
+    @Inject
+    Instance<RolesManager> rolesManagerServices;
+    
+    private UsersManager usersManagementService;
+    private GroupsManager groupsManagementService;
+
+    @PostConstruct
+    public void initialize() throws Exception {
+
+        // Look for the Service Provider implementation class targeted for users management.
+        usersManagementService = (UsersManager) getService("Users management", ENV_USERS_MANAGER_SERVICE, ENV_DEFAULT_USERS_MANAGER_SERVICE);
+        if (usersManagementService != null) {
+            try {
+                ContextualManager m = (ContextualManager) usersManagementService;
+                m.initialize(this);
+            } catch (ClassCastException e) {
+                // Manager is not contextual.
+            }    
+        }
+        
```

Comment:
Will add a log when no user or group management service is available, but i think it's not necessary when the classcast exception occurs, as the service is available as well, but not initialized and destroyed in a given context.... but it should work as well